app/src/main/java/io/appwrite/almostnetflix/feed/MovieDataSource.kt

```diff
@@ -56,27 +65,31 @@ class MovieDataSource(
     }
 
     suspend fun getMyWatchlist(userId: String, movieIds: List<String>): List<Movie> {
-        val movies = database.listDocuments(
+        val movies = databases.listDocuments(
+            databaseId = Configuration.DATABASE_ID,
             collectionId = Configuration.MOVIE_COLLECTION_ID,
             queries = listOf(
-                Query.equal("\$id", movieIds)
+                Query.equal("\$id", value = movieIds.toString()),
             )
         )
         return movies.convertTo(::Movie)
     }
 
     suspend fun addToWatchlist(userId: String, movieId: String): Boolean {
         try {
-            database.createDocument(
+            databases.createDocument(
+                databaseId = Configuration.DATABASE_ID,
                 collectionId = Configuration.WATCHLIST_COLLECTION_ID,
-                documentId = "unique()",
+                documentId = unique(),
                 data = mapOf(
                     "userId" to userId,
                     "movieId" to movieId,
                     "createdAt" to System.currentTimeMillis()
                 ),
-                read = listOf("user:$userId"),
-                write = listOf("user:$userId")
+                permissions = listOf(
+                    Permission.read("user:$userId"),
+                    Permission.write("user:$userId")
```

Comment:
We can use the role helpers here too:

```suggestion
                    Permission.read(Role.user(userId)),
                    Permission.write(Role.user(userId))
```